docs(api): liquid meniscus in API 2.23 (#17971)
OpenTrons/opentrons-api: api/docs/v2/basic_commands/liquids.rst

    depth = plate["A1"].bottom(z=2) # tip is 2 mm above well bottom
    pipette.aspirate(200, depth)

> Use the :py:meth:`~.Labware.Well.mensicus` method to aspirate from the meniscus of liquid in a well with a Flex pipette. First, you'll need to define the volume of liquid in your well: 

Reviewer: "relative to the meniscus"? "from" sounds like the exact location of the meniscus (which is not the usual behavior).

- Specify your starting liquid volume with :py:meth:`~.Labware.load_liquid`, or
- Use :py:meth:`~.InstrumentContext.measure_liquid_height` to define liquid height in a well before aspirating, or earlier in the protocol. 

    pipette.pick_up_tip()
    pipette.measure_liquid_height(plate["A2"])
    pipette.aspirate(200, location=plate["A2"].meniscus(target="end", z=-1)) # aspirates at 1 mm below the liquid meniscus

> The liquid meniscus changes when you aspirate liquid from a well. Set ``target`` to the ending position of the liquid within a well to ensure the pipette stays submerged while aspirating. For more information, see :ref:`well-meniscus`.

Reviewer: We'll sometimes shorten things by putting the parameter and value together in running text: 
```suggestion
The liquid meniscus changes when you aspirate liquid from a well. Set ``target=end`` to ensure the pipette stays submerged while aspirating. For more information, see :ref:`well-meniscus`.
```

    depth = plate["A1"].bottom(z=2) # tip is 2 mm above well bottom
    pipette.aspirate(200, depth)

> Use the :py:meth:`~.Well.meniscus` method to aspirate relative to the meniscus of liquid in a well with a Flex pipette. First, you'll need to define the volume of liquid in your well: 

Reviewer: We include the class in the previous paragraph, so let's do it here too.
```suggestion
Use the :py:meth:`.Well.meniscus` method to aspirate relative to the meniscus of liquid in a well with a Flex pipette. First, you'll need to determine the amount of liquid in your well one of two ways: 
```

> - Specify your starting liquid volume with :py:meth:`~.Labware.load_liquid`, or
> - Use :py:meth:`~.InstrumentContext.measure_liquid_height` to define liquid height in a well before aspirating, or earlier in the protocol. 

Reviewer: ```suggestion

- Specify your starting liquid volume with :py:meth:`~.Labware.load_liquid`.
- Measure the height of liquid with :py:meth:`~.InstrumentContext.measure_liquid_height`. 
```

Author: Leaving in `before aspirating, or earlier in the protocol` because this is important context for when to use each method, but switching it around to be less clunky.

Author: Actually, the code block below demonstrates this, and the length of this sentence is still bothering me. Removed.

> 

Reviewer: Needs something here to introduce the code block, like `This example measures the liquid height in well A2 of a plate and then immediately aspirates below the meniscus::`

    pipette.pick_up_tip()
    pipette.measure_liquid_height(plate["A2"])

>     pipette.aspirate(200, location=plate["A2"].meniscus(target="end", z=-1)) # aspirates at 1 mm below the liquid meniscus

Reviewer: `black` formatting (name all args, shorter lines to fit in code blocks)
```suggestion
    pipette.aspirate(
        volume=200, location=plate["A2"].meniscus(target="end", z=-1)
    )  # aspirates at 1 mm below the liquid meniscus
```

The liquid meniscus changes when you aspirate liquid from a well. Set ``target="end"`` to ensure the pipette stays submerged while aspirating. For more information, see :ref:`well-meniscus`.

> ``measure_liquid_height()`` works best with a new pipette tip each time. To save time and tips throughout your protocol, use ``Labware.load_liquid`` to specify starting liquid volumes.

Reviewer 2: If we do` load_liquid()` then we don't need to do `measure_liquid_height()` or `liquid_presence_detection=True`. As long as we load liquid (and the user is confident in their volumes) and `.meniscus()` it will work

Reviewer: ```suggestion
``measure_liquid_height()`` works best with a new pipette tip each time. To save time and tips throughout your protocol, use ``Labware.load_liquid`` instead to specify starting liquid volumes.
```

    depth = plate["B1"].bottom(z=2) # tip is 2 mm above well bottom
    pipette.dispense(200, depth)

Use the :py:meth:`.Well.meniscus` method to dispense at the meniscus of liquid in your well with a Flex pipette. First, you'll need to define the volume of liquid in your well: 
- Specify your starting liquid volume with :py:meth:`~Labware.load_liquid_by_well`, or
- Use :py:meth:`.measure_liquid_height` to define liquid height in a well before aspirating, or earlier in the protocol. 

    pipette.measure_liquid_height(plate["B1"])
    pipette.dispense(200, location=plate["B1"].meniscus(z=-1)) #dispenses 1 mm below the liquid meniscus

> The API will raise an error if no liquid is present in the well. 

Author: is this true? the API raises an error if no liquid is present to measure in the well?

Reviewer: Looks like this was fixed in #17957

    depth = plate["B1"].bottom(z=2) # tip is 2 mm above well bottom
    pipette.dispense(200, depth)

> Use the :py:meth:`~.Well.meniscus` method to dispense at the meniscus of liquid in your well with a Flex pipette. First, you'll need to define the volume of liquid in your well: 
> - Specify your starting liquid volume with :py:meth:`~.Labware.load_liquid`, or
> - Use :py:meth:`~.InstrumentContext.measure_liquid_height` to define liquid height in a well before aspirating, or earlier in the protocol. 
> 
>     pipette.measure_liquid_height(plate["B1"])
>     pipette.dispense(200, location=plate["B1"].meniscus(target="start", z=-1)) #dispenses 1 mm below the liquid meniscus

Reviewer: Same RST syntax and Python code style changes as above.
```suggestion
Use the :py:meth:`~.Well.meniscus` method to dispense at the meniscus of liquid in your well with a Flex pipette. First, you'll need to determine the amount of liquid in your well one of two ways: 
- Specify your starting liquid volume with :py:meth:`~.Labware.load_liquid`.
- Measure the height of liquid with :py:meth:`~.InstrumentContext.measure_liquid_height`. 

pipette.measure_liquid_height(plate["B1"])
pipette.dispense(
    volume=200, location=plate["B1"].meniscus(target="start", z=-1)
)  # dispenses 1 mm below the liquid meniscus
```

> The liquid meniscus changes when you dispense liquid into a well. Set ``target`` to the starting position of the liquid within a well to ensure the pipette begins the dispense at the liquid meniscus. For more information, see :ref:`well-meniscus`.

Reviewer: `target="start"` like we did for aspirate.

> ``measure_liquid_height()`` works best with a new pipette tip each time. To save time and tips throughout your protocol, use ``Labware.load_liquid`` to specify starting liquid volumes.

Reviewer: Add "instead" here if added above in aspirate section.

You can also require liquid presence for all aspirations performed with a given pipette. See :ref:`lpd`.

.. versionadded:: 2.20

Measure Liquids
===============

> The :py:meth:`.measure_liquid_height` method tells a Flex pipette to measure the height of liquid in a well. It returns the height of liquid in the well in mm. When ``measure_liquid_height()`` finds an empty well, it raises and error and pauses the protocol to let you resove the problem. 

Author: double check that this is 1) Flex-only and 2) requires a tip to measure liquid. fix "resove" typo in my next commit

You can also require liquid presence for all aspirations performed with a given pipette. See :ref:`lpd`.

.. versionadded:: 2.20

Measure Liquids
===============

The :py:meth:`~.InstrumentContext.measure_liquid_height` method tells a Flex pipette to measure the height of liquid relative to the bottom of a well. When ``measure_liquid_height()`` finds an empty well, it raises and error and pauses the protocol to let you resove the problem. 

> ``measure_liquid_height`` is a standalone method that records the height of liquid in a well during a protocol. You can use the liquid height to aspirate or dispense from, or move to, the liquid meniscus, either immediately after or later in your protocol.

Reviewer: ``measure_liquid_height()`` with parentheses 🐍 🧐
